common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -601,7 +601,7 @@ def get_demand_hint(self, hint_index):
             'hint_index': hint_index
         }
 
-    def get_problem_html(self, encapsulate=True):
+    def get_problem_html(self, encapsulate=True, save_notification_present=False, notification_message=None):
```

Comment:
I'm also going to need to pass hint text into this method (for a rough WIP, see https://github.com/edx/edx-platform/commit/fe04e7dcf3364ed3ae09b0380d61c56fcb23ddb6). So instead of having 2 variables here, can you just make a single "save_notification_message" variable and use the presence of that to determine whether or not the save notification is present?